src/Components/ExternalResult/ResultList.tsx

```diff
@@ -15,12 +14,16 @@ import PhoneNumberFormField from "../Form/FormFields/PhoneNumberFormField";
 import CountBlock from "../../CAREUI/display/Count";
 import { AdvancedFilterButton } from "../../CAREUI/interactive/FiltersSlideover";
 import Page from "../Common/components/Page";
+import routes from "../../Redux/api";
+import useQuery from "../../Utils/request/useQuery";
+import { IExternalResult } from "./models";
 
 const Loading = lazy(() => import("../Common/Loading"));
 
 export default function ResultList() {
-  const dispatch: any = useDispatch();
-  const [data, setData] = useState([]);
+  const [resultListData, setResultListData] = useState<
+    Partial<IExternalResult>[]
+  >([]);
   const [isLoading, setIsLoading] = useState(false);
   const [totalCount, setTotalCount] = useState(0);
```

Comment:
You don't need a state here again.

```suggestion
  const totalCount = data ? data.count : 0
```